integration_tests/CMakeLists.txt

```diff
@@ -434,6 +434,7 @@ RUN(NAME array_expr_05            LABELS cpython llvm c)
 RUN(NAME array_expr_06            LABELS cpython llvm c)
 RUN(NAME array_expr_07            LABELS cpython llvm c)
 RUN(NAME array_expr_08            LABELS cpython llvm c)
+RUN(NAME array_expr_09            LABELS cpython llvm c)
```

Comment:
We should also add the test case discussed in https://github.com/lcompilers/lpython/pull/2611#discussion_r1529321057 as an integration test.